XenModel/XenAPI-Extensions/Address.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace XenAPI
+{
+    public static class Address
+    {
+        public static List<string> FindIpAddresses(Dictionary<string, string> networks, string device)
+        {
+            if (networks == null || string.IsNullOrWhiteSpace(device))
+                return new List<string>();
+
+            // PR-1373 - VM_guest_metrics.networks is a dictionary of IP addresses in the format:
+            // [["0/ip", <IPv4 address>], 
+            //  ["0/ipv4/0", <IPv4 address>], ["0/ipv4/1", <IPv4 address>],
+            //  ["0/ipv6/0", <IPv6 address>], ["0/ipv6/1", <IPv6 address>]]
+
+            return
+                (from network in networks
+                    where network.Key.StartsWith(string.Format("{0}/ip", device))
+                    orderby network.Key
+                    select network.Value.Split(new[] {"\n", "%n"}, StringSplitOptions.None)).SelectMany(x => x).Distinct().ToList();

```

Comment:
I think it's safer to use StringSplitOptions.RemoveEmptyEntries; let's also add a relevant test.